File: 382-linked-list-random-node/linked-list-random-node.py
import random

class Solution:

    def __init__(self, head: Optional[ListNode]):
        self.head = head

    def getRandom(self) -> int:
        result = self.head.val
        curr = self.head.next
        i = 2

        while curr:
            if random.randint(1, i) == 1:
                result = curr.val

            curr = curr.next
            i += 1

        return result

File: 382-linked-list-random-node/linked-list-random-node.java
import java.util.Random;

/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode(int val) { this.val = val; }
 * }
 */

class Solution {

    private ListNode head;
    private Random random;

    public Solution(ListNode head) {
        this.head = head;
        this.random = new Random();
    }

    public int getRandom() {
        int result = head.val;
        ListNode curr = head.next;
        int count = 2;

        while (curr != null) {
            if (random.nextInt(count) == 0) {
                result = curr.val;
            }
            curr = curr.next;
            count++;
        }

        return result;
    }
}